### PR TITLE
GAWB-2891 fix thurloe swagger

### PR DESCRIPTION
**Pull request by zarsky-broad**
- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [ ] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [ ] Tell the tech lead (TL) that the PR exists if they wants to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via Slack and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

**Review by MatthewBemis — APPROVED** [review]
awesome!

File: build.sbt [hunk 1]
```diff
@@ -25,6 +25,7 @@ resolvers ++= Seq(
   "Broad Artifactory Snapshots" at "https://broadinstitute.jfrog.io/broadinstitute/libs-snapshot/")
 
 libraryDependencies ++= Seq(
+  "org.webjars" % "swagger-ui" % "2.2.5",
```
Comment (rushtong) [inline build.sbt]: It looks like there is a version 3.5.0 on webjars.org. Any reason not to use that?

File: src/main/resources/swagger/thurloe.yaml [hunk 2]
```diff
@@ -22,77 +22,67 @@ paths:
...
-                  "key": "string",
-                  "value": "string"
-                }
-              }]
+            application/json:
+              -
+                userId: 123
+                keyValuePair:
+                  key: foo
+                  value: bar
```
Comment (rushtong) [inline src/main/resources/swagger/thurloe.yaml:72]: Nice job switching over from json - devops' scans should be happier if they ever check Thurloe's swagger.

File: build.sbt (hunk 1 above)
Comment (zarsky-broad) [reply build.sbt]: I'm using the same version as orch uses. The 3.x versions are a totally new interface, so I'm punting on that for now.

File: src/main/scala/thurloe/service/ThurloeServiceActor.scala [hunk 3]
```diff
@@ -1,32 +1,86 @@
...
+            |        operationsSorter: "alpha",
+          """.stripMargin
 
+        HttpEntity(ContentType(MediaTypes.`text/html`),
+          indexHtml
+            .replace("your-client-id", authConfig.getString("googleClientId"))
+            .replace("scopeSeparator: \",\"", "scopeSeparator: \" \"")
+            .replace("jsonEditor: false,", "jsonEditor: false," + swaggerOptions)
+            .replace("url = \"http://petstore.swagger.io/v2/swagger.json\";",
+              "url = '/thurloe.yaml';"))
```
Comment (rushtong) [inline src/main/scala/thurloe/service/ThurloeServiceActor.scala:82]: These `replace`'s could be why you didn't choose the higher swagger version. IIRC, there is a significant change in the linked resources when upgrading.

File: build.sbt [hunk 4]
```diff
@@ -25,6 +25,7 @@ resolvers ++= Seq(
...
   "org.broadinstitute" %% "lenthall" % lenthallV,
```
Comment (davidangb) [inline build.sbt]: bah, I was hoping we could get rid of lenthall! It's still used to initialize the server. Oh well.

Comment (zarsky-broad) [reply build.sbt]: Yeah, unfortunately.

**Review by davidangb — APPROVED** [review]
:+1: at TL level. I did not test; please make sure another reviewer tests and looks at syntax.

File: build.sbt (hunk 1 above)
Comment (rushtong) [reply build.sbt]: Sounds good. There is a bug in the one that orch uses that has something to do with file uploads or downloads, but that's moot for Thurloe.

**Review by rushtong — APPROVED** [review]
👍  LGTM

**Comment by rushtong** [comment]
As a side note, I did test locally and it works.

File: build.sbt (hunk 1 above)
Comment (davidangb) [reply build.sbt]: I hate the 3.x interface, and it's a nontrivial upgrade

**Pull request merged**